src/Transform/ONNX/ConstProp.cpp

```diff
@@ -843,6 +848,27 @@ Value ConstPropReshape(
       .getResult();
 }
 
+//===----------------------------------------------------------------------===//
+// Code to perform constant propagation for ConstantOfShape.
+//===----------------------------------------------------------------------===//
+
+Value ConstPropConstantOfShape(PatternRewriter &rewriter, Value replacingValue,
+    Value shape, Attribute value) {
+  ConstPropCounters::count("ConstantOfShape", {shape});
+  llvm::SmallVector<int64_t, 4> shapeVector;
+  ElementsAttr shapeAttr =
+      getONNXConstantOp(shape).getValueAttr().cast<ElementsAttr>();
+  for (mlir::IntegerAttr dim : shapeAttr.getValues<IntegerAttr>())
+    shapeVector.push_back(dim.getInt());
+
+  ElementsAttr constElements = value.cast<ElementsAttr>();
```

Comment:
please add a comment here or in ConstProp.td explaining that we rely on value always being set, namely by the (questionable) "canonicalization" in ONNXConstantOfShapeOp::inferShapes()

(the casual reader my wonder about this given that the onnx spec says that the value attribute is optional)